Output when compiling the test with b234e44

<details>

<summary> Errors </summary>

```rs
rustc +stage1 ../bisect-test
error: concrete type differs from previous defining opaque type use
 --> ../bisect-test:9:10
  |
9 |     |_x| 15
  |          ^^ expected `impl Debug`, got `i32`
  |
note: previous use here
 --> ../bisect-test:9:5
  |
9 |     |_x| 15
  |     ^^^^^^^

error[E0720]: cannot resolve opaque type
 --> ../bisect-test:8:28
  |
8 | fn f<T>() -> impl Fn(T) -> impl Debug {
  |                            ^^^^^^^^^^ cannot resolve opaque type

error: concrete type differs from previous defining opaque type use
  --> ../bisect-test:25:10
   |
25 |     |_x| 15
   |          ^^ expected `impl Debug`, got `i32`
   |
note: previous use here
  --> ../bisect-test:25:5
   |
25 |     |_x| 15
   |     ^^^^^^^

error[E0720]: cannot resolve opaque type
  --> ../bisect-test:24:37
   |
24 | fn h<T>() -> impl Fn<(T,), Output = impl Debug> {
   |                                     ^^^^^^^^^^ cannot resolve opaque type

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0720`.`

```

</details>

Now it's a matter of deciding what the test name should be.

--- a/tests/ui/impl-trait/impl-fn-rpit-opaque-107883.rs
+++ b/tests/ui/impl-trait/impl-fn-rpit-opaque-107883.rs
@@ -1,0 +1,37 @@
+//@ check-pass
+// Regression test for <https;//github.com/rust-lang/rust/issues/107883>
+#![feature(impl_trait_in_fn_trait_return)]
+#![feature(unboxed_closures)] // only for `h`
+
+use std::fmt::Debug;
+
+fn f<T>() -> impl Fn(T) -> impl Debug {
+    |_x| 15
+}
+
+fn g<T>() -> impl MyFn<(T,), Out = impl Debug> {
+    |_x| 15
+}
+
+trait MyFn<T> {
+    type Out;
+}
+
+impl<T, U, F: Fn(T) -> U> MyFn<(T,)> for F {
+    type Out = U;
+}
+
+fn h<T>() -> impl Fn<(T,), Output = impl Debug> {
+    |_x| 15
+}
+
+fn f_<T>() -> impl Fn(T) -> impl Debug {
+    std::convert::identity(|_x| 15)
+}
+
+fn f__<T>() -> impl Fn(T) -> impl Debug {
+    let r = |_x| 15;
+    r
+}
+
+fn main() {}

